homeassistant/components/weather/__init__.py

```diff
@@ -1088,6 +1117,53 @@ async def async_get_forecast_service(
     }
 
 
+async def async_get_forecast_extreme_service(
+    weather: WeatherEntity, service_call: ServiceCall
+) -> ServiceResponse:
+    """Get weather forecast for specified attribute."""
+    forecast_attribute = service_call.data["attribute"]
+    forecast_limit_hours = service_call.data.get("limit_hours")
+    forecast_min_max = service_call.data["min_max"]
+
+    supported_features = weather.supported_features or 0
+    forecast_type = "hourly"
+    if (supported_features & WeatherEntityFeature.FORECAST_HOURLY) == 0:
+        raise_unsupported_forecast(weather.entity_id, forecast_type)
+    native_forecast_list = await weather.async_forecast_hourly()
+    if native_forecast_list is None:
+        result = None
+    else:
+        # pylint: disable-next=protected-access
+        converted_forecast_list = weather._convert_forecast(native_forecast_list)
+
+        # Check if forecast_limit_hours is given
+        if forecast_limit_hours is not None:
+            # Should not be greater than the length of the list
+            if forecast_limit_hours > len(converted_forecast_list):
+                # Cut list length according to forecast_limit_hours
+                converted_forecast_list = converted_forecast_list[:forecast_limit_hours]
+
+        # Check if key forecast_attribute is in list
+        if forecast_attribute not in converted_forecast_list[0]:
+            raise HomeAssistantError(
+                f"Forecast attribute '{forecast_attribute}' does not exist in forecast"
+            )
+
+        # Filter list to only get the key in forecast_attribute
+        converted_forecast_list = [
+            forecast[forecast_attribute] for forecast in converted_forecast_list
+        ]
+
+        # Get the min or max value
+        if forecast_min_max == "min":
+            result = min(converted_forecast_list)
```

Comment:
I'm not sure we should include calculations in the service parameters. I think calculations should be kept in a template as different calculations may be wanted by different users. Calculations are not dependent on the service result, but could be anything. Filtering is more dependent on the the service and service result, so is more acceptable as service parameters.

Reply:
Agree. The output can contain both min and max. Possibly also avg. WDYT?

Reply:
No, I don't think we should have any calculations in services.

Reply:
My idea was to retrieve the extreme values via a service and make all my sensors with templates redundant. If we do not use the calculation in services, this service is useless.

We could implement the hour filter in the normal forecast service instead.

Reply:
Removed calculations and renamed service